src/settings_table.cpp

```diff
@@ -500,9 +501,40 @@ static void ChangeTimekeepingUnits(int32_t)
 		UpdateAllServiceInterval(0);
 	}
 
+	/* If we are using calendar timekeeping, "minutes per year" must be default. */
+	if (!TimerGameEconomy::UsingWallclockUnits(true)) {
+		_settings_newgame.economy.minutes_per_calendar_year = CalendarTime::DEF_MINUTES_PER_YEAR;
+	}
+
 	InvalidateWindowClassesData(WC_GAME_OPTIONS, 0);
 }
 
+/**
+ * Callback after the player changes the minutes per year.
+ * @param Unused.
+ */
+static void ChangeMinutesPerYear(int32_t)
+{
+	/* We don't allow setting Minutes Per Year below default, unless it's to 0 for frozen calendar time. */
+	if (_game_mode == GM_MENU) {
+		if (_settings_newgame.economy.minutes_per_calendar_year < CalendarTime::DEF_MINUTES_PER_YEAR) {
+			_settings_newgame.economy.minutes_per_calendar_year = CalendarTime::FROZEN_MINUTES_PER_YEAR;
+		}
+	} else {
+		if (_settings_game.economy.minutes_per_calendar_year < CalendarTime::DEF_MINUTES_PER_YEAR) {
+			_settings_game.economy.minutes_per_calendar_year = CalendarTime::FROZEN_MINUTES_PER_YEAR;
+		}
+	}
```

Comment:
The problem here is, I think, that once it is 0, you can't get it off 0 again? Can't you use the `new_value` (which is the parameter) to figure out which direction we are going?

Honestly, I need to test this, I am just guessing behaviour by looking at code. Terrible way to judge functionality.

Reply:
Your guess was correct, as was your suggestion. 🙂